meson.build

```diff
@@ -0,0 +1,74 @@
+project('lcm', 'c', 'cpp',
+  version : '1.5.0',
+  default_options : ['default_library=static',
+                     'buildtype=release',
+                     'c_std=gnu99',
+                     'warning_level=0',
+                     'c_args=-Wno-format-zero-length'])
+
+# Dependencies
+glib_dep = dependency('glib-2.0')
+
+# set windows dependencies
+unix = host_machine.system() != 'windows'
+windows = not unix
+if windows
+  cpp = meson.get_compiler('cpp')
+  wsock32 = cpp.find_library('wsock32')
+  ws2_32 = cpp.find_library('ws2_32')
+endif
+
+# Core modules
+# lcm-gen compiles with warnings that cannot be disabled when lcm is used
+# as a Meson subproject in a parent project with more warnings enabled. Allow
+# parent projects to disable compilation of lcm-gen to fix warnings as needed.
```

Comment:
The decision to leave put the comment where it is now is to avoid confusion about whether lcm-gen is a core module of lcm. If we put the comment right above the `lcm_enable_lcmgen`, it would separate it from the core modules comment.